chart/kubeapps/templates/_helpers.tpl

```diff
@@ -25,10 +25,93 @@ If release name contains chart name it will be used as a full name.
 {{- end -}}
 
 {{/*
-Render image reference
+Return the proper apprepository image name
 */}}
-{{- define "kubeapps.image" -}}
-{{ .registry }}/{{ .repository }}:{{ .tag }}
```

Comment:
can't you just use a helper to return the repository? Then you can use it as:

```
{{ .registry }}/{{ template "kubeapps.repository" .repository }}:{{ .tag }}
```

with that you wouldn't need all the code below nor the changes in the templates

Reply:
+1 I'm not sure why we've extracted this to multiple helpers, when we can just use one. If the problem is that we need to pass in the global value, we can solve that by using a list. For example:

```
{{ template "kubeapps.image" (list .Values.apprepository.image .Values.global) }}
```

and then:

```
{{- define "kubeapps.image" -}}
{{- $image := index . 0 -}}
{{- $global := index . 1 -}}
{{- $registryName := $image.registry -}}
{{- if $global }}
    {{- if $global.regristry }}
        {{- $registryName := $global.registry -}}
    {{- end -}}
{{- end -}}
{{ .registryName }}/{{ .repository }}:{{ .tag }}
{{- end -}}
```

Reply:
Yes, the reason was related to passing the global value, I adapted your solution and it works fine